common/org.wso2.developerstudio.eclipse.general.project/src/org/wso2/developerstudio/eclipse/general/project/dialogs/DeveloperStudioElementProviderDialog.java

```diff
@@ -286,6 +288,18 @@ public void widgetSelected(SelectionEvent e) {
 		return container;
 	}
 
+	/**
+	 * This method passes a empty list. Sub classes should override this method to return list of files as required
+	 *
+	 * @param typesList
+	 * @return
+	 */
+	protected List<Object> addResourseFilesToList(List<Class<?>> typesList) {
```

Comment:
Shouldn't typesList be included in the JavaDocs ? Also it might be nice to describe what this collection will hold